smack-integration-test/src/main/java/org/jivesoftware/smack/roster/RosterIntegrationTest.java

```diff
@@ -104,7 +122,372 @@ private void checkIfAddedAndSubscribed(Collection<Jid> addresses) {
         }
         finally {
             rosterTwo.removeSubscribeListener(subscribeListener);
+            rosterOne.removeRosterListener(rosterListener);
+        }
+    }
+
+    /**
+     * Asserts that when a user sends out a presence subscription request, the server sends a roster push back to the
+     * user.
+     *
+     * @throws Exception when errors occur
+     */
+    @SmackIntegrationTest(section = "3.1.2", quote =
+        "After locally delivering or remotely routing the presence subscription request, the user's server MUST then " +
+        "send a roster push to all of the user's interested resources, containing the potential contact with a " +
+        "subscription state of \"none\" and with notation that the subscription is pending (via an 'ask' attribute " +
+        "whose value is \"subscribe\").")
+    public void testRosterPushAfterSubscriptionRequest() throws Exception {
+        IntegrationTestRosterUtil.ensureBothAccountsAreNotInEachOthersRoster(conOne, conTwo);
+        Roster.getInstanceFor(conTwo).setSubscriptionMode(Roster.SubscriptionMode.manual); // prevents a race condition when asserting the captured roster entry.
+        final ResultSyncPoint<RosterEntry, Exception> added = new ResultSyncPoint<>();
+
+        final RosterListener rosterListener = new AbstractRosterListener() {
+            @Override
+            public void entriesAdded(Collection<Jid> addresses) {
+                for (Jid jid : addresses) {
+                    if (!jid.equals(conTwo.getUser().asBareJid())) {
+                        continue;
+                    }
+                    final BareJid bareJid = conTwo.getUser().asBareJid();
+                    RosterEntry rosterEntry = rosterOne.getEntry(bareJid);
+                    added.signal(rosterEntry);
+                    return;
+                }
+            }
+        };
+        rosterOne.addRosterListener(rosterListener);
+
+        final Presence subscribe = PresenceBuilder.buildPresence()
+                .ofType(Presence.Type.subscribe)
+                .to(conTwo.getUser().asBareJid())
+                .build();
+
+        try {
+            conOne.sendStanza(subscribe);
+
+            final RosterEntry rosterEntry = assertResult(added, 2 * connection.getReplyTimeout(), "Expected the server to send a roster push back to '" + conOne.getUser() + "' after they sent a presence subscription request to '" + conTwo.getUser().asBareJid() + "' (but the server did not).");
```

Comment:
Out of curiosity, why do we use `2 * connection.ReplyTimeout()` here?

Ideally, there should be a code comment explaining the rationale why we derive from the default, or we should use the default.